### PR TITLE
Only override name of core api version

### PR DESCRIPTION
**Pull request by timja**
Fixes https://github.com/kubernetes-sigs/kustomize/issues/5072

Any feedback much appreciated I haven't contributed to this project before

Thanks!

### CODE REVIEW

File: api/krusty/namespaces_test.go [hunk 1]
```diff
@@ -801,3 +801,31 @@ metadata:
...
+// Test for issue #5072
+func TestNameNotOveriddenForNonCoreApiVersionOnANamespaceKind(t *testing.T) {
+	th := kusttest_test.MakeHarness(t)
+
+	th.WriteF("azure-servicebus.yaml", `
+apiVersion: servicebus.azure.com/v1beta20210101preview
+kind: Namespace
+metadata:
+  name: core-sb-99
+  namespace: podinfo
```
Comment (koba1t) [inline api/krusty/namespaces_test.go:815]: Please change the namespace name to without `podinfo`. 
I think this test case can't check to change `metadata.namespace`.

**Review by koba1t — CHANGES_REQUESTED** [review]

File: api/krusty/namespaces_test.go (hunk 1 above)
Comment (timja) [reply api/krusty/namespaces_test.go]: done

**Review by natasha41575 — COMMENTED** [review]
Thanks for the change! Pulled your branch and it is working great.

/lgtm
/approve

Please feel free to reach out (on slack or at one of the kustomize bug scrubs) if you are interested in finding more work to pick up. We are always happy to have new contributors!

**Pull request merged**